tools/uart_hw_test/uart_test_windows.c

```diff
@@ -0,0 +1,312 @@
+/*******************************************************************
+*
+* Copyright 2022 Samsung Electronics All Rights Reserved.
+*
+* File Name : trap.c
+* Description: Receive RAM and/or Userfs dump using UART
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+* either express or implied. See the License for the specific
+* language governing permissions and limitations under the License.
+*
+******************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/time.h>
+#include <windows.h>
+
+/****************************************************************************
+ * Preprocessor definitions
+ ****************************************************************************/
+#define UART_PATH_STR		"\\\\.\\COM%s"
+#define UART_PATH_STR_BELOW_TEN		"COM%s"
+#define TTYPATH_LEN		25
+#define MAX_PORTNO_LENGTH 	   10
+
+#define UART_TEST_COUNT		100
+
+#define UART_TEST_START_VAL	0xaa
+#define UART_TEST_VAL		0x55
+#define UART_TEST_STOP_VAL	0xbb
+
+#define UART_MAX_DELAY_MS	80
+#define UART_POLL_TIMEOUT_MS	2000
+
+#define HANDSHAKE_INTERVAL	100	/* msecs */
```

Comment:
Minor: please expand the comment to milliseconds since msecs is not a standard abbreviation and may be confusing.

Reply:
updated